initial github ci blog post (#348)
yast/yast.github.io: _posts/2023-02-28-github-ci.md

---
layout: post
date: 2023-02-28 12:00:00 +00:00
title: Systemd Container and Podman in Github CI
description: How to run Systemd containers and/or podman runtime on github CI
permalink: blog/2023-02-28/systemd-podman-github-ci
tags:
- Programming
- Systemd
- D-Installer
- github-ci
- podman
---

As D-Installer consists of several components like dbus backend, CLI or web frontend,
we see a need for test ideally in CI that each component can properly start and
communicate each other. For it we use a testing framework and more importantly
github CI where we need systemd container which in the end is not documented at all.
In following paragraphs we would like to share with you how we did it so

> anyone of you can inspire from it or use it for his/her/its (hello co-pilot) own project.

Reviewer: ```suggestion
anyone of you can inspire from it or use it for their own project.
```

a *singular they* is a lesser distraction
